frame/support/procedural/src/pallet/parse/event.rs

```diff
@@ -202,14 +209,14 @@ impl EventDef {
 				let args = variant.fields.iter()
 					.map(|field| {
 						metadata.iter().find(|m| m.0 == field.ty)
-							.map(|m| m.1.clone())
+							.cloned()
 							.unwrap_or_else(|| {
-								clean_type_string(&field.ty.to_token_stream().to_string())
+								(field.ty.clone(), clean_type_string(&field.ty.to_token_stream().to_string()))
```

Comment:
(related to the comment above, I'm not sure the string here is very valuable, I feel like the type is what user want, no more)